src/pyhf/optimize/mixins.py

```diff
@@ -0,0 +1,137 @@
+    ):
+
+        minimizer = self._get_minimizer(
+            func, x0, bounds, fixed_vals=fixed_vals, do_grad=do_grad
+        )
+        result = self._minimize(
+            minimizer,
+            func,
+            x0,
+            do_grad=do_grad,
+            bounds=bounds,
+            fixed_vals=fixed_vals,
+            options=options,
+        )
+
+        try:
+            assert result.success
+        except AssertionError:
+            log.error(result)
+            raise
+        return result
+
+    def _internal_postprocess(self, fitresult, stitch_pars):
+        """
+        Post-process the fit result, ``scipy.optimize.OptimizeResult`` object.
+
+        Returns:
+            fitresult (`scipy.optimize.OptimizeResult`): A modified version of the fit result.
+        """
+        tensorlib, _ = get_backend()
+
+        fitted_pars = stitch_pars(tensorlib.astensor(fitresult.x))
+        # extract number of fixed parameters
+        num_fixed_pars = len(fitted_pars) - len(fitresult.x)
+
+        # check if uncertainties were provided
+        uncertainties = getattr(fitresult, 'unc', None)
+        if uncertainties is not None:
+            # stitch in zero-uncertainty for fixed values
+            fitted_uncs = stitch_pars(
+                tensorlib.astensor(uncertainties),
+                stitch_with=tensorlib.zeros(num_fixed_pars),
+            )
+            fitresult.unc = fitted_uncs
+            fitted_pars = tensorlib.stack([fitted_pars, fitted_uncs], axis=1)
+
+        fitresult.x = fitted_pars
+        fitresult.fun = tensorlib.astensor(fitresult.fun)
+
+        return fitresult
+
+    def minimize(
+        self,
+        objective,
+        data,
+        pdf,
+        init_pars,
+        par_bounds,
+        fixed_vals=None,
+        return_fitted_val=False,
+        return_fit_object=False,
+        do_grad=False,
+        do_stitch=False,
+        **kwargs,
+    ):
+        """
+        Find parameters that minimize the objective.
+
+        Args:
+            objective (`func`): objective function
+            data (`list`): observed data
+            pdf (`pyhf.pdf.Model`): model
+            init_pars (`list`): initial parameters
+            par_bounds (`list`): parameter boundaries
+            fixed_vals (`list`): fixed parameter values
+            return_fitted_val (`bool`): return bestfit value of the objective
+            return_fit_object (`bool`): return ``scipy.optimize.OptimizeResult``
+            do_grad (`bool`): enable autodifferentiation mode. Default is off.
+            do_stitch (`bool`): enable splicing/stitching fixed parameter.
+            kwargs: other options to pass through to underlying minimizer
+
+        Returns:
+            parameters (`tensor`): fitted parameters
+            minimum (`float`): if ``return_fitted_val`` flagged, return minimized objective value
+            result (scipy.optimize.OptimizeResult`): if ``return_fit_object`` flagged
```

Comment:
This doesn't render out cleanly in Sphinx at the moment. Can we use the [style used in `hypotest`](https://github.com/scikit-hep/pyhf/blob/d35f4b487d3a03d42edb6f3a49122283fedd5d59/src/pyhf/infer/__init__.py#L48)?